x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSessionCCSUtils.java

```diff
@@ -23,14 +23,30 @@
 import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
 
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-class CcsUtils {
+class EsqlSessionCCSUtils {
 
-    private CcsUtils() {}
+    private EsqlSessionCCSUtils() {}
+
+    // visible for testing
+    static Map<String, FieldCapabilitiesFailure> determineUnavailableRemoteClusters(List<FieldCapabilitiesFailure> failures) {
+        Map<String, FieldCapabilitiesFailure> unavailableRemotes = new HashMap<>();
+        for (FieldCapabilitiesFailure failure : failures) {
+            if (ExceptionsHelper.isRemoteUnavailableException(failure.getException())) {
+                for (String indexExpression : failure.getIndices()) {
+                    if (indexExpression.indexOf(RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR) > 0) {
```

Comment:
Are index expressions already parsed here? E.g. remote/index wildcards, etc.?

Reply:
Yes. The is called in IndexResolver after calling field-caps. It sifts through the FieldCapsResponse failure list. 
Remember this is not new code. I just moved this method out of IndexResolver into the utils method, as Costin had requested in my earlier PR.